index.ts

```diff
@@ -53,58 +79,115 @@ const corsPost = cors((req, callback) => {
 app.use(favicon(path.join(__dirname, "favicon.ico")));
 app.get("/", (req, res) => res.sendFile(`${process.cwd()}/index.html`));
 
-app.get("/api/v2/:key", corsGet, async (req, res) => {
-  try {
-    const key = req.params.key;
-    const file = bucket.file(key);
-    await file.getMetadata();
-    res.status(200);
-    res.setHeader("content-type", "application/octet-stream");
-    file.createReadStream().pipe(res);
-  } catch (error) {
-    console.error(error);
-    res.status(404).json({ message: "Could not find the file." });
+app.get("/api/v2/:key", corsGet, async (req: any, res: any) => {
+  if (getStorageType() === "S3") {
+    try {
+      await (async () => {
+        function streamToString(stream: any) {
+          return new Promise(function (resolve, reject) {
+            const chunks: any = [];
+            stream.on("data", function (chunk: any) {
+              chunks.push(chunk);
+            });
+            stream.on("error", reject);
+            stream.on("end", function () {
+              resolve(Buffer.concat(chunks).toString("utf8"));
+            });
+          });
+        }
+
+        const key = req.params.key;
+
+        const command = new GetObjectCommand({
+          Bucket: process.env.SCALEWAY_BUCKET_NAME,
+          Key: key,
+        });
+
+        const { Body } = await client.send(command);
+        const bodyContents = await streamToString(Body);
+        console.log(bodyContents);
+      })();
+    } catch (error) {
+      console.error(error);
+      res.status(404).json({ message: "Could not find the file." });
```

Comment:
Si tu globalise le error handling, ça pourrait tout de même être sympa d'avoir une différentiation des erreurs entre S3 and GCS. Mais faut demander plus haut, je sais pas si c'est utile.